code/lib/cli/src/generators/baseGenerator.ts

```diff
@@ -187,6 +189,11 @@ export async function baseGenerator(
   const isStorybookInMonorepository = packageManager.isStorybookInMonorepo();
   const shouldApplyRequireWrapperOnPackageNames = isStorybookInMonorepository || pnp;
 
+  if (!builder) {
+    // eslint-disable-next-line no-param-reassign
+    builder = await detectBuilder(packageManager, projectType);
+  }
```

Comment:
Should we not ask for the framework instead?

Reply:
The workflow currently of sb init is to be able to pass a builder and a type, there's no framework option. The change I made was mostly a refactor, to fix a bug where, if you pass something like `--type=react` and no builder, you will get prompted to choose the builder only once, not twice. The other change is that if you pass `--type=angular` where there is only one builder, it won't ask, because it doesn't need to